service/models/user.go

```diff
@@ -0,0 +1,72 @@
+package models
+
+import (
+	"github.com/google/uuid"
+	"github.com/murilogilfelpeto/ps-tag-onboarding-go/service/models/exceptions"
+)
+
+type User struct {
+	id        string
+	firstName string
+	lastName  string
+	email     string
+	age       int
+}
+
+func NewUser(id string, firstName string, lastName string, email string, age int) (User, error) {
+	newUser := &User{
+		id:        id,
+		firstName: firstName,
+		lastName:  lastName,
+		email:     email,
+		age:       age,
+	}
+	if err := validateUser(newUser); err != nil {
+		return User{}, err
+	}
+	return *newUser, nil
+}
+
+func validateUser(user *User) error {
+	if user.id != "" && uuid.Validate(user.id) != nil {
+		return &exceptions.UserValidationErr{Message: "Invalid id"}
+	}
+
+	if user.firstName == "" {
+		return &exceptions.UserValidationErr{Message: "First name is required"}
+	}
+
+	if user.lastName == "" {
+		return &exceptions.UserValidationErr{Message: "Last name is required"}
+	}
+
+	if user.email == "" {
+		return &exceptions.UserValidationErr{Message: "email is required"}
+	}
+
+	if user.age < 18 {
+		return &exceptions.UserValidationErr{Message: "User must be at least 18 years old"}
+	}
+
+	return nil
+}
```

Comment:
suggestion: I would export the user validation func, and make it receiver of User struct. 
question: Isn't literal initialization of the user model sufficient? This is a straightforward struct that may not require a new func.

```go
u := model.User{Name: "muri", Age: "55", ...}
u.Validate()
fmt.Println( "Name: %s", u.Name)
```

Reply:
I made this way, so the user will be validated when it's build, to guarantee that every time you want an user, he'll be validated